### PR TITLE
samples: Central NUS fix handling data longer than 212 bytes

### PR DESCRIPTION
**Pull request by KAGA164**
This commit provides valid handling data longer than 212 bytes.
To handle this, one of the synchronization mechanisms is used.

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

### CODE REVIEW

**Comment by KAGA164** [comment]
@kapi-no @grochu Please review

File: samples/bluetooth/central_uart/src/main.c [hunk 1]
```diff
@@ -478,5 +492,10 @@ void main(void)
 			printk("Failed to send data over BLE connection"
 			       "(err %d)\n", err);
 		}
+
+		err = k_sem_take(&nus_write_sem, NUS_WRITE_TIMEOUT);
```
Comment (kapi-no) [inline samples/bluetooth/central_uart/src/main.c]: Are you sure that 150 timeout is enough? Did you consider using semaphore without any timeout restriction if ```ble_data_sent``` callback is reliable?

Comment (KAGA164) [reply samples/bluetooth/central_uart/src/main.c]: I choose this time experimentally for this data amount. I also considered taking a semaphore with `K_FOREVER` but I think if the Bluetooth hang, application can received UART messages and try to send it while heap memory is not full. Exceeding this time causes only displaying a message with timeout information.

**Comment by KAGA164** [comment]
@kapi-no @grochu I added hardware flow control for UART. Please look again on it.

**Comment by KAGA164** [comment]
@joerchan @grochu I realized that with hardware flow control we have to connect terminal because each printk call is blocking so without terminal we hang up on every printk call and wait for CTS signal forever. Is it ok ?

**Comment by joerchan** [comment]
@KAGA164 Opening the terminal must be done anyway. So it is fine by me. And passes tests as well.

**Comment by joerchan** [comment]
@grochu I would like your input on this before merging.

**Pull request merged**